packages/connect-agreement/src/thegraph/parsers/agreement.ts

```diff
@@ -0,0 +1,18 @@
+import { QueryResult } from '@aragon/connect-thegraph'
+
+import { AgreementData } from '../../types'
+
+export function parseAgreement(result: QueryResult): AgreementData {
+  const agreement = result.data.agreement
+
+  if (!agreement) {
```

Comment:
Should we start having validating functions for the data we receive from The Graph (e.g. doing `isValid(agreement)` rather than `!agreement`) in the `parsers/*` files? For example we rely on `agreement.currentVersion` to be an object below, but it might not be the case.

Reply:
Good point, I agree we can definitely do better here! Let's discuss in general what's the ideal pattern that could work for these types of validations. I'll be happy to address it in a separate PR :)

Reply:
Agreed, it would make sense to do this in all the app connectors at once :+1: